cowdao_cowpy/app_data/utils.py

```diff
@@ -96,9 +101,67 @@ def generate_app_data(
     if partner_fee:
         app_data_doc["metadata"]["partnerFee"] = (asdict(partner_fee),)
 
+    if app_code:
+        app_data_doc["appCode"] = app_code
     stringified_data = stringify_deterministic(app_data_doc)
     app_data_hash = keccak256(stringified_data.encode("utf-8"))
     return CreateAppData(
         AppDataObject(fullAppData=AppData(stringified_data)),
         AppDataHash(root=app_data_hash),
     )
+
+
+def check_app_data_exists(orderbook: OrderBookApi, app_data_hash: AppDataHash) -> bool:
+    try:
+        # Attempt to fetch the app data from the orderbook
+        asyncio.run(orderbook.get_app_data(app_data_hash))
+    except Exception as e:  # noqa
+        print(f"App data not found: {e}")
+        return False
+    return True
+
+
+def build_and_post_app_data(
+    orderbook: OrderBookApi,
+    app_code: str | None = None,
+    referrer_address: str | None = None,
+    partner_fee: PartnerFee | None = None,
+    graffiti: str | None = None,
+) -> str:
+    create_app_data = generate_app_data(
+        app_code=app_code,
+        referrer_address=referrer_address,
+        partner_fee=partner_fee,
+        graffiti=graffiti,
+    )
+    if not check_app_data_exists(orderbook, create_app_data.app_data_hash):
+        print("App data does not exist, uploading... to ", orderbook.config.chain_id)
+        asyncio.run(
+            orderbook.put_app_data(
+                create_app_data.app_data_object, create_app_data.app_data_hash
+            )
+        )
+    return create_app_data.app_data_hash.root
+
+
+def build_all_app_codes(
+    env: Envs = "prod",
+    app_code: str = "cowdao_cowpy",
```

Comment:
Hmm i shuold really mooove this to constants maybe, 

perhaps @mfw78 can suggest some better defaults?